pkg/api/annotation.go

```diff
@@ -0,0 +1,73 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/pyroscope-io/pyroscope/pkg/model"
+	"github.com/pyroscope-io/pyroscope/pkg/server/httputils"
+	"github.com/pyroscope-io/pyroscope/pkg/util/attime"
+	"github.com/sirupsen/logrus"
+)
+
+type AnnotationsService interface {
+	CreateAnnotation(ctx context.Context, params model.CreateAnnotation) (*model.Annotation, error)
+}
+type AnnotationsCtrl struct {
+	log       *logrus.Logger
+	svc       AnnotationsService
+	httpUtils httputils.Utils
+}
+
+func NewAnnotationsCtrl(log *logrus.Logger, svc AnnotationsService, httpUtils httputils.Utils) *AnnotationsCtrl {
+	return &AnnotationsCtrl{
+		log:       log,
+		svc:       svc,
+		httpUtils: httpUtils,
+	}
+}
+
+type CreateParams struct {
+	AppName   string `json:"appName"`
+	Timestamp int64  `json:"timestamp"`
+	Content   string `json:"content"`
+}
+
+func (ctrl *AnnotationsCtrl) CreateHandler(w http.ResponseWriter, r *http.Request) {
+	var params CreateParams
+	if err := json.NewDecoder(r.Body).Decode(&params); err != nil {
+		ctrl.httpUtils.WriteInternalServerError(r, w, err, "failed to unmarshal JSON")
```

Comment:
Please use `HandleError` instead, it sets proper HTTP code and creates proper response body:
https://github.com/pyroscope-io/pyroscope/blob/90f21bf9d254c442968662b4ee2699cf172537f5/pkg/server/httputils/default.go#L37-L46